python/ray/autoscaler/launch_and_verify_cluster.py

```diff
@@ -229,7 +231,49 @@ def cleanup_cluster(cluster_config):
     raise last_error
 
 
-def run_ray_commands(cluster_config, retries, no_config_cache, num_expected_nodes=1):
+def cleanup_security_group(ec2_client, id):
+    retry = 0
+    while retry < 10:
+        try:
+            ec2_client.delete_security_group(GroupId=id)
+            return
+        except botocore.exceptions.ClientError as e:
+            if e.response["Error"]["Code"] == "DependencyViolation":
+                sleep_time = 2**retry
+                print(
+                    f"Waiting {sleep_time}s for the instance to be terminated before deleting the security group {id}"  # noqa E501
+                )
+                time.sleep(sleep_time)
+                retry += 1
+            else:
+                print(f"Error deleting security group: {e}")
+                return
+
+
+def cleanup_security_groups(config):
+    try:
+        ec2_client = boto3.client("ec2", region_name="us-west-2")
+        response = ec2_client.describe_security_groups(
+            Filters=[
+                {
+                    "Name": "tag-key",
+                    "Values": ["ray-autoscaler"],
```

Comment:
Nit, is it better to import and use the `RAY` constant here, in case its value changes?